src/api/useChangeDone.tsx

```diff
@@ -0,0 +1,23 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { todoListQueryKeys } from './useGetTodoList';
+import { getDb, updateDb } from './mockDb';
+
+const changeDoneTodo = async (id: string) => {
+  const todo = getDb().find((todo) => todo.id === id);
+  if (!todo) {
+    throw new Error('Todo not found');
+  }
+  todo.done = !todo.done; // toggle done status of the todo
+  updateDb(todo);
```

Comment:
Use this please instead of the mutation above:
```suggestion
  updateDb({ ...todo, done: !todo.done });
```